### ISSUE TITLE
Release Build image and tags issues

### ISSUE THREAD

**Issue by mandarjog** [istio/istio#13602]
There are several issues with image releases at present.

- [ ] Pre-release images are pushed to docker.io as tagged version while qualification testing is in progress. If qualification testing fails, the bits of the published image have to be re uploaded. We should not release these images prematurely. Instead they should be placed in a `gcr.io/istio-prerelase/` bucket. After the release is qualified, 
    - docker images should be promoted from gcr.io to docker.io 
    - build artifcats should move from gsc buckets to github releases, with updated image names.
- [ ] Link stamped version of released build shows `dirty-Modified`. This should be fixed.  example:
```
Istio Mixer: root@host-docker.io/istio-1.1.4-1cc5778b17a3c949db11951395cbf2a168e30696-dirty-Modified
```
- [ ] Stats gauge (`pkg/version/stats`) uses buildInfo.GitTag which is produced by `git describe`. Which that to using `b.Version + "-" + b.GitRevision`. That way build explicitly controls the version.

**Comment by howardjohn** [comment]
```
$ dk run -it gcr.io/istio-release/pilot:release-1.2-20190524-22-12 version
version.BuildInfo{Version:"release-1.2-20190524-22-12", GitRevision:"0b01a44b4fe3b87908ca4301d804c236c8ced587-dirty", User:"root", Host:"07b44639-7e71-11e9-a0d5-0a580a2c0304", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.6-876-g0b01a44"}
```

Need to make sure this is fixed as well, likely the same problem. See GitTag = 1.1.6 for the 1.2 release

**Comment by howardjohn** [comment]
I am planning to fix this by 1.4

**Comment by howardjohn** [comment]
This is fixed now in 1.4 daily builds, and 1.4 will be built with the same code so will be fixed in that. other tracking in #16887
